File: src/main/java/mil/nga/tiff/compression/UnsupportedCompression.java
package mil.nga.tiff.compression;

import mil.nga.tiff.util.TiffException;

import java.nio.ByteOrder;

/**
 * Unsupported compression
 *
 * @author michaelknigge
 */
public class UnsupportedCompression implements CompressionDecoder, CompressionEncoder {

    private final String message;

    /**
     * Constructor
     *
     * @param message message of the TiffException
     * @since 2.0.1
     */
    public UnsupportedCompression(final String message) {
        this.message = message;
    }

    @Override
    public byte[] decode(byte[] bytes, ByteOrder byteOrder) {
        throw new TiffException(this.message);
    }

    @Override
    public boolean rowEncoding() {
        return false;
    }

    @Override
    public byte[] encode(byte[] bytes, ByteOrder byteOrder) {
        throw new TiffException(this.message);
    }
}

File: src/main/java/mil/nga/tiff/compression/UnsupportedCompression.kt
package mil.nga.tiff.compression

import mil.nga.tiff.util.TiffException
import java.nio.ByteOrder

/**
 * Unsupported compression
 */
class UnsupportedCompression(private val message: String) : CompressionDecoder, CompressionEncoder {
    override fun decode(bytes: ByteArray, byteOrder: ByteOrder): ByteArray {
        throw TiffException(this.message)
    }

    override fun rowEncoding(): Boolean {
        return false
    }

    override fun encode(bytes: ByteArray, byteOrder: ByteOrder): ByteArray {
        throw TiffException(this.message)
    }
}
